webviz_subsurface/plugins/_simulation_time_series/views/subplot_view/_settings/_time_series.py

```diff
@@ -0,0 +1,373 @@
+            ),
+            html.Button(
+                "Vector Calculator",
+                id=self.register_component_unique_id(
+                    TimeSeriesSettings.Ids.VECTOR_CALCULATOR_OPEN_BUTTON
+                ),
+                style={
+                    "margin-top": "5px",
+                    "margin-bottom": "5px",
+                },
+            ),
+            wcc.Dialog(
+                title="Vector Calculator",
+                id=self.register_component_unique_id(
+                    TimeSeriesSettings.Ids.VECTOR_CALCULATOR_DIALOG
+                ),
+                draggable=True,
+                max_width="lg",
+                children=[
+                    html.Div(
+                        style={"height": "60vh"},
+                        children=[
+                            wsc.VectorCalculator(
+                                id=self.register_component_unique_id(
+                                    TimeSeriesSettings.Ids.VECTOR_CALCULATOR
+                                ),
+                                vectors=self._vector_calculator_data,
+                                expressions=self._predefined_expressions,
+                            )
+                        ],
+                    )
+                ],
+            ),
+            dcc.Store(
+                id=self.register_component_unique_id(
+                    TimeSeriesSettings.Ids.VECTOR_CALCULATOR_EXPRESSIONS
+                ),
+                data=self._predefined_expressions,
+            ),
+            dcc.Store(
+                id=self.register_component_unique_id(
+                    TimeSeriesSettings.Ids.VECTOR_CALCULATOR_EXPRESSIONS_OPEN_DIALOG
+                ),
+                data=self._predefined_expressions,
+            ),
+            dcc.Store(
+                # NOTE:Used to trigger graph update callback if data has
+                # changed, i.e. no change of regular INPUT html-elements
+                id=self.register_component_unique_id(
+                    TimeSeriesSettings.Ids.GRAPH_DATA_HAS_CHANGED_TRIGGER
+                ),
+                data=0,
+            ),
+        ]
+
+    def set_callbacks(self) -> None:
+        @callback(
+            Output(
+                self.component_unique_id(
+                    TimeSeriesSettings.Ids.VECTOR_CALCULATOR_DIALOG
+                ).to_string(),
+                "open",
+            ),
+            [
+                Input(
+                    self.component_unique_id(
+                        TimeSeriesSettings.Ids.VECTOR_CALCULATOR_OPEN_BUTTON
+                    ).to_string(),
+                    "n_clicks",
+                ),
+            ],
+            [
+                State(
+                    self.component_unique_id(
+                        TimeSeriesSettings.Ids.VECTOR_CALCULATOR_DIALOG
+                    ).to_string(),
+                    "open",
+                )
+            ],
```

Comment:
Why `Input` and `State` in a list? Saw this multiple places, you might want to check that you follow one syntax, preferrably the newer one.